lib/Data/ValueProxy.js

```diff
@@ -231,14 +230,15 @@ export class ValueProxy extends Events {
      */
     static fromModel(obj, key) {
         return new ValueProxy({
+            generation: obj._generation,
```

Comment:
Accessing private fields outside of a class is not such a good idea. You need either make generation a public field or add a `getGeneration()` method.